### PR TITLE
Add PlayerFlowerPotManipulateEvent

### PR DESCRIPTION
**Pull request by MisterVector**
Added an event for when a player places an item in or takes an item out of a flower pot.

This is a revisit to an old PR I had submitted nearly 3 years ago but wasn't able to finish because of an issue with server-side code that handles event cancellation.

### CODE REVIEW

**Comment by fgRuslan** [comment]
:sunflower:

**Comment by fgRuslan** [comment]
:white_flower: :tulip:

**Comment by Proximyst** [comment]
Please don't spam comments.

**Comment by fgRuslan** [comment]
Sorry

File: Spigot-API-Patches/0185-Add-PlayerFlowerPotManipulateEvent.patch [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+Subject: [PATCH] Add PlayerFlowerPotManipulateEvent
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/event/player/PlayerFlowerPotManipulateEvent.java b/src/main/java/com/destroystokyo/paper/event/player/PlayerFlowerPotManipulateEvent.java
+new file mode 100644
+index 00000000..4f4b728a
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/event/player/PlayerFlowerPotManipulateEvent.java
+@@ -0,0 +1,75 @@
++package com.destroystokyo.paper.event.player;
```
Comment (Proximyst) [inline Spigot-API-Patches/0185-Add-PlayerFlowerPotManipulateEvent.patch]:
```suggestion
+package io.papermc.paper.event.player;
```

File: Spigot-Server-Patches/0411-Implement-PlayerFlowerPotManipulateEvent.patch [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
++
++                return true;
++            }
++            // Paper end
++
+             if (flag1) {
+                 world.setTypeAndData(blockposition, block.getBlockData(), 3);
+                 entityhuman.a(StatisticList.POT_FLOWER);
+-- 
+2.20.1.windows.1
```
Comment (Proximyst) [inline Spigot-Server-Patches/0411-Implement-PlayerFlowerPotManipulateEvent.patch]: My condolences 🤓

**Review by Proximyst — DISMISSED** [review]
Needs a rebase, but lgtm.

**Comment by MisterVector** [comment]
Alright rebased on top of latest. Had to change one line of code server-side due to latest server changes. Should be good to go!

**Review by zachbr — DISMISSED** [review]

**Comment by Proximyst** [comment]
@limbo-app ready

**Review by limbo-app — APPROVED** [review]
Pull request automatically approved for merging - a core team member has previously approved this pull request.

**Pull request merged**